workspaces/marketplace/plugins/marketplace/src/components/MarketplacePluginContent.tsx

```diff
@@ -196,42 +235,115 @@ export const MarketplacePluginContent = ({
   const highlights = plugin.spec?.highlights ?? [];
 
   const pluginActionButton = () => {
+    const disablePluginActions =
+      pluginConfigPerm.data?.read !== 'ALLOW' &&
+      pluginConfigPerm.data?.write !== 'ALLOW';
+
+    if (disablePluginActions) {
+      return (
+        <Tooltip title="You don't have permission to install plugins or view their configurations. Contact your administrator to request access or assistance.">
+          <div>
+            <Button
+              color="primary"
+              variant="contained"
+              disabled={disablePluginActions}
+              data-testId="install-disabled"
+            >
+              Install
+            </Button>
+          </div>
+        </Tooltip>
+      );
+    }
+
+    if (
+      plugin.spec?.installStatus === MarketplacePluginInstallStatus.Installed &&
+      enableActionsButtonFeature
+    ) {
+      return (
+        <>
+          <Button
+            variant="contained"
+            disableElevation
+            onClick={handleClick}
+            endIcon={<KeyboardArrowDownIcon />}
+            color="primary"
+            data-testId="plugin-actions"
+          >
+            Actions
+          </Button>
+          <ActionsMenu
+            id="actions-button"
+            data-testId="actions-button"
+            MenuListProps={{
+              'aria-labelledby': 'actions-button',
+            }}
+            anchorEl={anchorEl}
+            open={open}
+            onClose={handleClose}
+          >
+            {/* Comment out the Edit menu, if edit is not functional */}
+            <MenuItem onClick={handleEdit} disableRipple>
+              <ListItemIcon
+                style={{ paddingLeft: '10px', paddingRight: '8px' }}
+              >
+                <EditIcon fontSize="small" />
+              </ListItemIcon>
+              <ListItemText primary="Edit" secondary="Plugin configurations" />
+            </MenuItem>
+            {/* Make the appropriate API call to check the plugin status and show Enable/Disable action accordingly */}
+            <MenuItem
+              data-testId={isPluginEnabled ? 'disable-plugin' : 'enable-plugin'}
+            >
+              <ListItemIcon>
+                <Switch
+                  size="small"
+                  checked={isPluginEnabled}
+                  disableRipple
+                  onChange={handleToggle}
+                  inputProps={{
+                    'aria-label': isPluginEnabled
+                      ? 'disable-plugin'
+                      : 'enable-plugin',
+                  }}
+                />
```

Comment:
We can use
```
import ToggleOffOutlinedIcon from '@mui/icons-material/ToggleOffOutlined';
```
and
```
import ToggleOnOutlinedIcon from '@mui/icons-material/ToggleOnOutlined';
```
instead of `Switch`, which is same as the design in Figma. This way we can handle margin around icons in a better consistent way. Now because both icons are in different sizes, they have different paddings/margins.

Reply:
I used the `Switch` component on purpose to show the enable to disable (and vise versa) transition. With the icons it won't be possible
@ShiranHi what are your thoughts on this ?